app/infrastructure/persistence/migrations/env.py

```
@@ -0,0 +1,73 @@
+from sqlalchemy.engine import Connection
+from sqlalchemy.ext.asyncio import async_engine_from_config
+
+from app.core.config import get_settings
+from app.infrastructure.persistence.database import Base
+
+settings = get_settings()
+if settings.database_backend != "postgres":
+    print(
+        "Alembic: database_backend is not 'postgres' (current: %r). "
+        "Firestore has no SQL schema; skipping migrations."
+        % (settings.database_backend,),
```

Ruff (0.14.14)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)